### DISCUSSION TITLE
How do you select WinZip AES-256 encryption for Zip format?

### DISCUSSION THREAD

**Discussion by ra-dave** [rikyoz/bit7z#138] [Q&A] [answered]
Zip defaults to ZipCrypto encryption.  Is it possible to select WinZip AES-256 instead?  I'm looking through the code and it seems like this might not be supported.

**Comment by rikyoz** [comment, marked as answer]
Hi!
Changing the encryption algorithm of ZIP archives is "indirectly" supported only since the latest v4.0.0 RC via the `BitCompressor::setFormatProperty` method, which allows you to specify any format property using the same parameters you can use with 7-zip (e.g., https://sevenzip.osdn.jp/chm/cmdline/switches/method.htm): 

```cpp
Bit7zLibrary lib{};
BitFileCompressor compressor( lib, BitFormat::Zip );
compressor.setFormatProperty( L"em", L"AES256" );
compressor.setPassword( "password" );
compressor.compressFile( "path/to/file.txt", "path/to/archive.zip" );
```

Please note that the `setFormatProperty` method supports only wide char C strings, hence the need for the `L"..."` (in contrast with the rest of the library).

Unfortunately, at the moment, bit7z doesn't provide a more user-friendly API for changing the encryption algorithm of ZIP archives. Still, I'll definitely work on adding one in a future version of the library!

**Comment by ra-dave** [comment]
Thanks, that works well!  It's so simple I'm not sure I would worry about adding something specifically to your API to make it more user-friendly.  Perhaps just documentation (if you don't already have some for this).

Very nice library, BTW!

**Reply by rikyoz** [reply]
You're welcome!

> It's so simple I'm not sure I would worry about adding something specifically to your API to make it more user-friendly. Perhaps just documentation (if you don't already have some for this).

Well, as long as it doesn't worsen the library's code, I think adding it shouldn't take too much effort. I created this library to be user-friendly, after all!
But yeah, I'll evaluate for sure if it's the case to add it.

> Very nice library, BTW!

Thank you!
